Documentation/test-suites-overview.md

```diff
@@ -0,0 +1,240 @@
+
+At Mender we have different test suites that follow the Development Lifecycle,
+verifying our software at different levels through the different phases of the
+Development Lifecycle.
+
+Due to our microservices architecture, we name our test suites in a different
+way than standards defined by ISTQB (International Software Testing
+Qualifications Board) or other online references. See (ISTQB Foundation Level
+Syllabus)[https://www.istqb.org/downloads/syllabi/foundation-level-syllabus.html].
+
+Nevertheless, we follow QA best practices and principles in verifying our
+software product and its individual components at different levels.
+
+## Introduction: programming languages
+
+Except for the GUI and some client addons in C, the core components of Mender
+are written in Go.
+
+The unit-level tests are written in the same programming language as the code
+itself (usually Go), while for every other test suite we use the Python
+programming language and [Pytest](https://docs.pytest.org) framework.
+
+## Unit tests
+
+Test suite       | unit tests
+---------------- | ----------
+ISTQB equivalent | Component testing
+Test object      | The code inside a single repository
+Source code      | Repository specific, all `*_test.go` files
+How to run       | `go test ./...`
+Dependencies     | Some repositories (namely, backend services) require MongoDB server
+
+This is the lowest test level, where the code is closely inspected to exercise
+it in as many circumstances as possible. We collect coverage statistics that are
+published on each repository's GitHub page.
+
+## Backend acceptance tests
+
+Test suite       | backend acceptance tests
+---------------- | ----------
+ISTQB equivalent | Component integration testing
+Test object      | Interfaces of a single repository
+Source code      | Repository specific, `tests/` directory
+How to run       | See [mendertesting template](https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-docker-acceptance.yml) and [integration wrapper](https://github.com/mendersoftware/integration/blob/master/extra/travis-testing/run-test-environment) (example below)
+Dependencies     | Docker, docker-compose
+
+These test suites focus on the interactions between integrated components or, in
+our microservices architecture, services, and their interfaces.
+
+### Running locally
+
+Acceptance tests are run in a separate container within the same network as the
+service under test and other services when required. Each repository has a
+Docker compose file (usually named `docker-compose-acceptance.yml`) describing
+other services, networks, and aliases that are required to run the tests.
+
+This separate container for the tests is built independently for all
+repositories in
+[mender-test-containers](https://github.com/mendersoftware/mender-test-containers/tree/master/backend-acceptance-testing)
+and published to Docker Hub as
+`mendersoftware/mender-test-containers:acceptance-testing`.
```

Comment:
unfortunately this is still only aspirational as of today...

Reply:
Well, this is the only way to run acceptance tests for, for example, `deviceauth`.

I didn't want to document the two procedures and then explain which repo uses which, do you think we should?

Reply:
no... that was just a reminder that we're unfortunately not there yet - I was very much hoping to get to the migration of some more repos this sprint... but tests